src/test/unit/status_line/status_line_message_timer.test.ts

```diff
@@ -0,0 +1,130 @@
+import { strict as assert } from "assert";
+
+import { StatusLineMessageTimer } from "../../../status_line/status_line_message_timer";
+import { TimerFunctions } from "../../../utils";
+
+/**
+ * Simulates a set of timer functions by keeping track of what timers are created and cleared, and allowing manual
+ * firing of known timers.
+ *
+ * The ids are always incrementing.
+ */
+class ManualTimerContainer implements TimerFunctions<number> {
```

Comment:
You'll notice this is duplicated from `Timer`. I chose to do this, rather than putting it in a `utils` function, as the repetition really isn't that bad and also means that each class's test suite can change behavior as needed, without worrying about external impact. That said, I waffled on this quite a bit so I'm not tied to it

Reply:
Disregard this